src/common/loggingEvents.ts

```diff
@@ -18,18 +18,24 @@ export class UserSignInTriggered implements BaseEvent {
 
 export class UserSignInCompleted implements BaseEvent {
     type = EventType.UserSignInCompleted;
-    constructor(public correlationId: string, public succeeded: boolean) { }
+    constructor(public correlationId: string, public succeeded: boolean, public retrievedFromCache: boolean) { }
 }
 
 export class UserSignInSucceeded extends UserSignInCompleted {
     constructor(public correlationId: string, public credential: Credential) {
-        super(correlationId, true);
+        super(correlationId, true, false);
     }
 }
 
 export class UserSignInFailed extends UserSignInCompleted {
     constructor(public correlationId: string, public err: Error) {
-        super(correlationId, false);
+        super(correlationId, false, false);
+    }
+}
+
+export class CredentialRetrievedFromLocalCredentialManager extends UserSignInCompleted {

```

Comment:
No need to have `CredentialRetrievedFromLocalCredentialManager` at all?
just use `UserSignInSucceeded` event and check `retrievedFromCache` from the event would be enough.
And even more, `UserSignInSucceeded` and `UserSignInFailed` are needed, if we want to reduce the event types. 
Use `UserSignInCompleted` everywhere, and check on `succeeded` and `retrievedFromCache`, it would meets what we need

Reply:
OK, but keep the `UserSignInSucceeded ` and `UserSignInFailed ` since the property is different

---
In reply to: [384317701](https://github.com/docascode/vscode-docs-build/pull/70#discussion_r384317701) [](ancestors = 384317701)